datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -0,0 +1,128 @@
+package chaindatafetcher
+
+import (
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/event"
+	"github.com/klaytn/klaytn/log"
+	"github.com/klaytn/klaytn/networks/p2p"
+	"github.com/klaytn/klaytn/networks/rpc"
+	"github.com/klaytn/klaytn/node"
+	"sync"
+)
+
+var logger = log.NewModuleLogger(log.ChainDataFetcher)
+
+type ChainDataFetcher struct {
+	config *ChainDataFetcherConfig
+
+	blockchain *blockchain.BlockChain
+
+	chainCh  chan blockchain.ChainEvent
+	chainSub event.Subscription
+
+	reqCh  chan uint64 // TODO-ChainDataFetcher add logic to insert new requests from APIs to this channel
+	resCh  chan uint64
+	stopCh chan struct{}
+
+	numHandlers int
+
+	wg sync.WaitGroup
+}
+
+func NewChainDataFetcher(ctx *node.ServiceContext, cfg *ChainDataFetcherConfig) (*ChainDataFetcher, error) {
+	return &ChainDataFetcher{
+		config:      cfg,
+		chainCh:     make(chan blockchain.ChainEvent, cfg.BlockChannelSize),
+		reqCh:       make(chan uint64, cfg.RequestChSize),
+		resCh:       make(chan uint64, cfg.ResponseChSize),
+		stopCh:      make(chan struct{}),
+		numHandlers: cfg.NumHandlers,
+	}, nil
+}
+
+func (f *ChainDataFetcher) Protocols() []p2p.Protocol {
+	return []p2p.Protocol{}
+}
+
+func (f *ChainDataFetcher) APIs() []rpc.API {
+	// TODO-ChainDataFetcher add APIs to start or stop chaindata fetcher
+	return []rpc.API{}
+}
+
+func (f *ChainDataFetcher) Start(server p2p.Server) error {
+	// launch multiple goroutines to handle new blocks
+	for i := 0; i < f.numHandlers; i++ {
+		f.wg.Add(1)
```

Comment:
This seems to be just a matter of taste, however, how about passing wait group as a parameter of handler and call `Add` and `Done` inside the handler? In that case, we can remove a wrapping function and just call `go f.handler(f.wg)`

Reply:
Right. Even `ChainDataFetcher` has the wait group variable, so no need to pass it as a parameter. Applied on a9ca7a8536ad08739a1a40fc515a807023667ece.